File: VRSampleApp/app/src/main/java/com/ooyala/sample/screen/MainActivity.java
package com.ooyala.sample.screen;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;

import com.ooyala.sample.R;
import com.ooyala.sample.fragmentfactory.FragmentFactory;
import com.ooyala.sample.interfaces.VideoChooseInterface;
import com.ooyala.sample.utils.VideoData;

import static com.ooyala.android.util.TvHelper.isTargetDeviceTV;


public class MainActivity extends AppCompatActivity implements VideoChooseInterface {

  private Toolbar toolbar;

  @Override
  protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.main_activity);
    toolbar = findViewById(R.id.toolbar);

    setupToolbar();
    showRecyclerFragment();

    getSupportFragmentManager().addOnBackStackChangedListener(new FragmentManager.OnBackStackChangedListener() {
      @Override
      public void onBackStackChanged() {
        if (getSupportActionBar() != null) {
          boolean showHomeAsUp = getSupportFragmentManager().getBackStackEntryCount() > 0;
          getSupportActionBar().setDisplayHomeAsUpEnabled(showHomeAsUp);
          if (!showHomeAsUp) {
            toolbar.setTitle(R.string.app_name);
          }
        }
      }
    });

    hideToolbarForTv();
  }

  private void hideToolbarForTv() {
    if (isTargetDeviceTV(this) && getSupportActionBar() != null) {
      getSupportActionBar().hide();
    }
  }

  @Override
  public void onBackPressed() {
    setupToolbar();
    FragmentManager supportFragmentManager = getSupportFragmentManager();

    if (supportFragmentManager.getBackStackEntryCount() >= 1) {
      supportFragmentManager.popBackStack();
    } else {
      super.onBackPressed();
    }
  }

  @Override
  public void onVideoChoose(VideoData data) {
    VideoFragment fragment = FragmentFactory.getFragmentByAdType(data.getAdType());
    fragment.setArguments(data);
    showVideoFragment(fragment);
    toolbar.setTitle(data.getTitle());
    toolbar.hideOverflowMenu();
  }

  public void showVideoFragment(VideoFragment fragment) {
    final FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
    fragmentTransaction.replace(R.id.container, fragment, VideoFragment.TAG).addToBackStack(null).commit();
  }

  private void showRecyclerFragment() {
    final FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
    fragmentTransaction.replace(R.id.container, new VideoRecyclerFragment(), VideoRecyclerFragment.TAG).commit();
  }

  private void setupToolbar() {
    setSupportActionBar(toolbar);
    getSupportActionBar().setTitle(R.string.app_name);
    getSupportActionBar().show();
    toolbar.bringToFront();
    toolbar.showOverflowMenu();

    toolbar.setNavigationOnClickListener(new View.OnClickListener() {
      @Override
      public void onClick(View v) {
        MainActivity.this.onBackPressed();
      }
    });
  }

  @Override
  public boolean onCreateOptionsMenu(Menu menu) {
    getMenuInflater().inflate(R.menu.menu, menu);
    return true;
  }

  @Override
  public boolean onOptionsItemSelected(MenuItem item) {
    if(item.getItemId() == R.id.menu_add_video){
      DialogFragment dialogFragment = new EmbedCodeDialogFragment();
      dialogFragment.show(getSupportFragmentManager(), EmbedCodeDialogFragment.class.getSimpleName());
    }
    return super.onOptionsItemSelected(item);
  }
}

File: VRSampleApp/app/src/main/java/com/ooyala/sample/screen/MainActivity.kt
package com.ooyala.sample.screen

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.ooyala.sample.R
import com.ooyala.sample.VideoFragment
import com.ooyala.sample.interfaces.ItemClickedInterface
import com.ooyala.sample.utils.VideoItemType
import com.ooyala.sample.utils.VideoData
import kotlinx.android.synthetic.main.main_activity.*

class MainActivity : AppCompatActivity(), ItemClickedInterface {

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.main_activity)
    setupToolbar()
    showRecyclerFragment()
    supportFragmentManager.addOnBackStackChangedListener { onBackStackChanged() }
  }

  override fun onBackPressed() {
    if (supportFragmentManager.backStackEntryCount >= 1) {
      supportFragmentManager.popBackStack()
    } else {
      super.onBackPressed()
    }
  }

  override fun onItemClicked(data: VideoData) {
    if (data.type == VideoItemType.VIDEO) {
      openVideoFragment(data)
    }
  }

  private fun openVideoFragment(data: VideoData) {
    val fragmentTransaction = supportFragmentManager.beginTransaction()
    fragmentTransaction.replace(R.id.container, VideoFragment(data), VideoFragment.TAG).addToBackStack(null).commit()
    toolbar.title = data.title
  }

  private fun showRecyclerFragment() {
    val fragmentTransaction = supportFragmentManager.beginTransaction()
    fragmentTransaction.replace(R.id.container, VideoRecyclerFragment(), VideoRecyclerFragment.TAG).commit()
    toolbar.title = getString(R.string.app_name)

  }

  private fun setupToolbar() {
    setSupportActionBar(toolbar)
    toolbar?.bringToFront()
    toolbar.setNavigationOnClickListener { onBackPressed() }
  }

  private fun onBackStackChanged() {
    supportActionBar?.setDisplayHomeAsUpEnabled(supportFragmentManager.backStackEntryCount > 0)
  }


  /*override fun completePlayerSetup(asked: Boolean) {
    if (asked) {
      val tvRatingConfiguration = FCCTVRatingConfiguration.Builder().setDurationSeconds(5).build()
      val options = Options.Builder()
              .setTVRatingConfiguration(tvRatingConfiguration)
              .setBypassPCodeMatching(true)
              .setUseExoPlayer(true)
              .build()
      player = OoyalaPlayer(pcode, PlayerDomain(domain), options)
      player.addObserver(this)

      val skinOptions = SkinOptions.Builder().build()
      val skinLayoutController = OoyalaSkinLayoutController(application, player_skin_layout, player, skinOptions)
      skinLayoutController.addObserver(this)
      player.embedCode = this.embedCode
    }
  }*/
}
